backend/src/main/java/com/jobwait/security/exchange/LinkedInOAuthExchange.java

```diff
@@ -0,0 +1,68 @@
+package com.jobwait.security.exchange;
+
+import java.util.UUID;
+
+import org.springframework.http.MediaType;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.jobwait.security.AuthToken;
+import com.jobwait.security.SecurityFaults;
+import com.jobwait.security.UUIDJumbler;
+import com.jobwait.security.linkedin.LinkedInUserProfile;
+import com.jobwait.util.Tuple;
+
+public final class LinkedInOAuthExchange extends OAuthExchange {
+    private static WebClient localClient = WebClient.create();
+
+    private static String codeToBearerToken(String authCode) {
+        // Everything except authCode should be pulled in from ENV
+        final String clientId = "";
+        final String clientSecret = "";
+        final String redirectURI = "";
+        final String linkedInAccessTokenURI = "https://www.linkedin.com/oauth/v2/accessToken";
+
+        MultiValueMap<String, String> formData = new LinkedMultiValueMap<>();
+        formData.add("grant_type", "authorization_code");
+        formData.add("code", authCode);
+        formData.add("client_id", clientId);
+        formData.add("client_secret", clientSecret);
+        formData.add("redirect_uri", redirectURI);
+
+        return localClient
+                .post()
+                .uri(linkedInAccessTokenURI)
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .bodyValue(formData)
+                .retrieve()
+                .bodyToMono(AuthToken.class)
+                .onErrorMap(err -> SecurityFaults.FailToGetTokenFault(err.getMessage()))
+                .block().authToken;
+    }
+
+    @Override
+    public String getUserID(String authCode) {
+        String bearerToken = codeToBearerToken(authCode);
+
+        String linkedInUserInfoURI = "https://api.linkedin.com/v2/userinfo";
+
+        String linkedInUserID = localClient
+                .get()
+                .uri(linkedInUserInfoURI)
+                .headers(h -> h.setBearerAuth(bearerToken))
+                .retrieve()
+                .bodyToMono(LinkedInUserProfile.class)
+                .onErrorMap(err -> SecurityFaults.FailToGetUserIDFault(err.getMessage()))
+                .block().sub;
+
+        return linkedInUserID;
+    }
+
+    @Override
+    public Tuple<String, String> getUserIDAndHash(String authCode) {
+        String userID = this.getUserID(authCode);
+        String hashString = UUIDJumbler.jumbleUUID(userID);
+        return new Tuple<String, String>(hashString, userID);
+    }
```

Comment:
Please swap the arguments in line 66, the order of the returned data is not consistent with the function name.

Reply:
Agreed, but I stopped messing with this function because its probably just going to return the hash anyways. I dont see a need to return the raw userID and hash. 

It should really just be `public String getHash(String authCode)`.

Reply:
ah very well then, I'll re-review it once that change has been made